### PR TITLE
Update help link and aka.ms references for RDG

### PR DESCRIPTION
**Pull request by captainsafia**
Part of https://github.com/dotnet/aspnetcore/issues/48184

### CODE REVIEW

File: src/Http/Http.Extensions/gen/Resources.resx [hunk 1]
```diff
@@ -121,22 +121,22 @@
...
+    <value>Unable to statically resolve route pattern for endpoint. Compile-time endpoint generation will skip this endpoint and the endpoint will be generated at runtime. For more information, please see https://aka.ms/rdg-known-issues</value>
   </data>
   <data name="UnableToResolveMethod_Title" xml:space="preserve">
     <value>Unable to resolve endpoint handler</value>
   </data>
   <data name="UnableToResolveMethod_Message" xml:space="preserve">
-    <value>Unable to statically resolve endpoint handler method. Only method groups, lambda expressions or readonly fields/variables are allowed. Compile-time endpoint generation will skip this endpoint and the endpoint will be generated at runtime. For more information, please see aka.ms/aot-known-issues</value>
+    <value>Unable to statically resolve endpoint handler method. Only method groups, lambda expressions or readonly fields/variables are allowed. Compile-time endpoint generation will skip this endpoint and the endpoint will be generated at runtime. For more information, please see aka.ms/rdg-known-issues</value>
```
Comment (JamesNK) [inline src/Http/Http.Extensions/gen/Resources.resx:130]:
```suggestion
    <value>Unable to statically resolve endpoint handler method. Only method groups, lambda expressions or readonly fields/variables are allowed. Compile-time endpoint generation will skip this endpoint and the endpoint will be generated at runtime. For more information, please see https://aka.ms/rdg-known-issues</value>
```

File: src/Http/Http.Extensions/gen/Resources.resx [hunk 2]
```diff
@@ -121,22 +121,22 @@
     <value>Unable to resolve route pattern</value>
   </data>
   <data name="UnableToResolveRoutePattern_Message" xml:space="preserve">
-    <value>Unable to statically resolve route pattern for endpoint. Compile-time endpoint generation will skip this endpoint and the endpoint will be generated at runtime. For more information, please see https://aka.ms/aot-known-issues</value>
+    <value>Unable to statically resolve route pattern for endpoint. Compile-time endpoint generation will skip this endpoint and the endpoint will be generated at runtime. For more information, please see https://aka.ms/rdg-known-issues</value>
```
Comment (JamesNK) [inline src/Http/Http.Extensions/gen/Resources.resx:124]: We tend to put our links under the `/aspnet/` segment on aka.ms. [For example.](https://github.com/search?q=repo%3Adotnet%2Faspnetcore%20https%3A%2F%2Faka.ms%2Faspnet%2F&type=code)

Change the link to `https://aka.ms/aspnet/rdg-known-issues`?

Comment (captainsafia) [reply src/Http/Http.Extensions/gen/Resources.resx]: Good point -- done!

File: src/Http/Http.Extensions/test/RequestDelegateGenerator/CompileTimeCreationTests.AsParameters.cs [hunk 3]
```diff
@@ -39,16 +39,16 @@ public static object[][] BadArgumentListActions
         get
         {
             static string GetAbstractTypeError(Type type)
-                => $"The abstract type '{TypeNameHelper.GetTypeDisplayName(type, fullName: false)}' is not supported.";
+                => $"The abstract type '{TypeNameHelper.GetTypeDisplayName(type, fullName: false)}' is not supported. For more information, please see https://aka.ms/aspnet/rdg-known-issues";
 
             static string GetMultipleContructorsError(Type type)
-                => $"Only a single public parameterized constructor is allowed for type '{TypeNameHelper.GetTypeDisplayName(type, fullName: false)}'.";
+                => $"Only a single public parameterized constructor is allowed for type '{TypeNameHelper.GetTypeDisplayName(type, fullName: false)}'. For more information, please see https://aka.ms/aspnet/rdg-known-issues";
```
Comment (adityamandaleeka) [inline src/Http/Http.Extensions/test/RequestDelegateGenerator/CompileTimeCreationTests.AsParameters.cs:45]: Will issues/info about each of these things be added to that known issues issue? Right now, there's no further information about this limitation at that link.

I guess the meta-question is, if the error is descriptive enough about what the problem is, is there value in adding the link?

Comment (captainsafia) [reply src/Http/Http.Extensions/test/RequestDelegateGenerator/CompileTimeCreationTests.AsParameters.cs:45]: > Will issues/info about each of these things be added to that known issues issue? Right now, there's no further information about this limitation at that link.

They'll be added to a diagnostics info page in the docs similar to what we have in [for analyzer diagnostics like this one](https://learn.microsoft.com/en-us/aspnet/core/diagnostics/asp0014?view=aspnetcore-7.0).

The aka.ms link currently directs to the old known issues GitHub page but the page on docs.microsoft.com will enumerate through more of the warnings.

**Review by JamesNK — APPROVED** [review]

File: src/Http/Http.Extensions/test/RequestDelegateGenerator/CompileTimeCreationTests.AsParameters.cs (hunk 3 above)
Comment (adityamandaleeka) [reply src/Http/Http.Extensions/test/RequestDelegateGenerator/CompileTimeCreationTests.AsParameters.cs:45]: Gotcha, thanks.

**Comment by adityamandaleeka** [comment]
Approved.

**Review by adityamandaleeka — APPROVED** [review]

**Comment by captainsafia** [comment]
@adityamandaleeka Mind hitting merge as well?

**Pull request merged**